### PR TITLE
test (gradle-plugin/it) : Add gradle integration test for ProjectLabelEnricher (#1262)

### LINKED ISSUE #1262: Gradle-Enricher: ProjectLabelEnricher
### Description

Child of #31

ProjectLabelEnricher works with Gradle

### AC
- [x] Enricher documentation is ported to Gradle #1331
- [x] Create IT to verify that `jkube.enricher.jkube-project-label.app` flag works as expected #1386
- [x] kubernetes.yml is generated as expected #1331
- [x] openshift.yml is generated as expected #1331

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Fix partially #1262 

+ Add integration test for ProjectLabelEnricher
+ Add ProjectLabelEnricher docs for Gradle Plugins as well 
Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [X] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [X] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ProjectLabelIT.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  public void k8sResource_whenRun_generatesK8sManifestsWithProjectLabels() throws IOException, ParseException {
+    // When
+    ITGradleRunner itGradleRunner = gradleRunner.withITProject("project-label");
+    if (expectedDir.equals("custom")) {
+      itGradleRunner.withArguments("-Pjkube.enricher.jkube-project-label.useProjectLabel=true",
+          "-Pjkube.enricher.jkube-project-label.provider=custom-provider",
+          "k8sResource", "--stacktrace");
+    } else {
+      itGradleRunner.withArguments("k8sResource", "--stacktrace");
+    }
```
Comment (manusa) [inline gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ProjectLabelIT.java:56]: Creating a parameterized test with an if-else condition inside checking for the parameter is NOT OK.

Comment (manusa) [reply gradle-plugin/it/src/test/java/org/eclipse/jkube/gradle/plugin/tests/ProjectLabelIT.java:56]: If there's common setup or assert logic you want to verify, then move that to a separate method and call it from the specific test.

However, in this case, what you probably want to provide is a different set of arguments for each parameter.

```
  @Parameterized.Parameter
  public String arguments;
```

**Review by manusa — APPROVED** [review]

File: gradle-plugin/it/src/it/project-label/expected/custom/kubernetes.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                      fieldPath: metadata.name
+              image: repository/project-label:latest
+              imagePullPolicy: IfNotPresent
+              name: repository-project-label
+              ports:
+                - containerPort: 8080
+                  name: http
+                  protocol: TCP
+              securityContext:
+                privileged: false
```
Comment (sunix) [inline gradle-plugin/it/src/it/project-label/expected/custom/kubernetes.yml:80]: minor: could you add new line ?

**Review by sunix — APPROVED** [review]

**Comment by manusa** [comment]
What's missing from #1262 ?

**Pull request merged**